src/plugins/dashboard/public/application/actions/copy_to_dashboard_modal.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useCallback, useState } from 'react';
+import { omit } from 'lodash';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFormRow,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiPanel,
+  EuiRadio,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { DashboardCopyToCapabilities } from './copy_to_dashboard_action';
+import { DashboardPicker } from '../../services/presentation_util';
+import { dashboardCopyToDashboardAction } from '../../dashboard_strings';
+import { EmbeddableStateTransfer, IEmbeddable } from '../../services/embeddable';
+import { createDashboardEditUrl, DashboardConstants } from '../..';
+
+interface CopyToDashboardModalProps {
+  capabilities: DashboardCopyToCapabilities;
+  stateTransfer: EmbeddableStateTransfer;
+  PresentationUtilContext: React.FC;
+  embeddable: IEmbeddable;
+  dashboardId?: string;
+  closeModal: () => void;
+}
+
+export function CopyToDashboardModal({
+  PresentationUtilContext,
+  stateTransfer,
+  capabilities,
+  dashboardId,
+  embeddable,
+  closeModal,
+}: CopyToDashboardModalProps) {
+  const [dashboardOption, setDashboardOption] = useState<'new' | 'existing'>('existing');
+  const [selectedDashboard, setSelectedDashboard] = useState<{ id: string; name: string } | null>(
+    null
+  );
+
+  const onSubmit = useCallback(() => {
+    const state = {
+      input: omit(embeddable.getInput(), 'id'),
+      type: embeddable.type,
+    };
+
+    const path =
+      dashboardOption === 'existing' && selectedDashboard
+        ? `#${createDashboardEditUrl(selectedDashboard.id, true)}`
+        : `#${DashboardConstants.CREATE_NEW_DASHBOARD_URL}`;
+
+    closeModal();
+    stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
+      state,
+      path,
+    });
+  }, [dashboardOption, embeddable, selectedDashboard, stateTransfer, closeModal]);
+
+  return (
+    <PresentationUtilContext>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{dashboardCopyToDashboardAction.getDisplayName()}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <>
+          <EuiText className="dshCopyToParagraph">
+            <p>{dashboardCopyToDashboardAction.getDescription()}</p>
+          </EuiText>
```

Comment:
Apologies the piecemeal review. We can do away with the remaing CSS change (margin-bottom) and this `className` by inserting an `<EuiSpacer /> ` below the EuiText.

Reply:
Makes sense, updated this to use the spacer